examples/PlaneCurve.jl

```diff
@@ -0,0 +1,273 @@
+module PlaneCurveModule
+using Oscar, Markdown
+import Base.:(==)
+
+export Point, ideal_point, AffinePlaneCurve, ProjPlaneCurve, hash, degree,
+       jacobi_ideal, curve_components, isirreducible, isreduced, reduction,
+       union
+
+################################################################################
+
+abstract type PlaneCurve end
+
+################################################################################
+# Point (not specific to curves).
+################################################################################
+# Point (first attempt).
+# The point is described by its coordinates.
+# ambient_dim gives the dimension of the space to which belongs the point.
+
+mutable struct Point{S}
+  coord::Array{S, 1}
+  ambient_dim::Int64
+  function Point(coordinates::Array{S, 1}) where {S <: FieldElem}
+    r = new{S}()
+    r.coord = coordinates
+    r.ambient_dim = length(coordinates)
+    return r
+  end
+end
+
+function Base.show(io::IO, P::Point)
+  println(io, "Point with coordinates ", P.coord)
+end
+
+################################################################################
+# Associate a maximal ideal to a point in a given ring (not specific to curves)
+
+@doc Markdown.doc"""
+    ideal_point(P::Point{S}, R::MPolyRing{S}) where S <: FieldElem
+
+Return the maximal ideal associated to the point `P` in the ring `R`.
+"""
+function ideal_point(R::MPolyRing{S}, P::Point{S}) where S <: FieldElem
+  V = gens(R)
+  return ideal(R, [V[i] - P.coord[i] for i in 1:length(V)])
+end
+
+################################################################################
+# Structure of Affine Plane Curves and Projective Plane Curves
+################################################################################
+
+mutable struct AffinePlaneCurve{S <: FieldElem} <: PlaneCurve
+  eq::Oscar.MPolyElem{S}                  # Equation of the curve (polynomial in two variables)
+  degree::Int64                           # degree of the equation of the curve
+  dimension::Int64                        # dimension of the curve (as a variety)
+  components::Dict
+  function AffinePlaneCurve(eq::Oscar.MPolyElem{S}) where {S <: FieldElem}
+    if nvars(parent(eq)) != 2
+       error("The defining equation must belong to a ring with two variables")
+    elseif isconstant(eq)
+       error("The defining equation must be non constant")
+    end

```

Comment:
You could use here and elsewhere this style of error checks (but the current one is also OK):
```suggestion
    nvars(parent(eq)) == 2 || error("The defining equation must belong to a ring with two variables")
    !isconstant(eq) || error("The defining equation must be non-constant")
```